### PR TITLE
push latest tag for docker images

### LINKED ISSUE #73: `latest` tag not updated
The [latest](https://hub.docker.com/layers/otel/opentelemetry-collector/latest/images/sha256-cec18c72535bebeaca3615e841b19066b468f87a2e29b724b118b5b7bf8808ce?context=explore) tag in dockerhub isn't getting updated during the release process. This is causing user confusion.

### PR DESCRIPTION
**Pull request by codeboten**
This is an alternative implementation to https://github.com/open-telemetry/opentelemetry-collector-releases/pull/76. I'll leave it up to reviewers on whether the bash script is easier to maintain than a github action, I don't have a strong preference here, but am not very familiar with typescript which is why i put together this alternative PR.

Fix #73

### CODE REVIEW

**Review by bogdandrutu — COMMENTED** [review]
How can we test this?

**Comment by codeboten** [comment]
> How can we test this?

I ran some test of the script locally, I can try to see if I can run the tests against a fork of this repo

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -48,3 +48,10 @@ jobs:
           args: release --rm-dist --timeout 1h
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Push latest Docker tag
+        run: |
+          ./scripts/push-latest.sh otel/opentelemetry-collector ${{ github.ref_name }}
```
Comment (jpkrohling) [inline .github/workflows/release.yaml:54]: Isn't there a way to add this to goreleaser's config? Like here:

https://github.com/open-telemetry/opentelemetry-collector-releases/blob/1af799780977b1ef24d628ed3665a64d704f6263/.goreleaser.yaml#L235-L239

Comment (mx-psi) [reply .github/workflows/release.yaml:54]: Based on https://goreleaser.com/customization/docker/#customization I think we can just add `otel/opentelemetry-collector:latest` in the appropriate place?

Comment (jpkrohling) [reply .github/workflows/release.yaml:54]: Yes, but note that the "docker" section is for the arch-specific images, whereas the "docker_manifests" is the meta image holding the arch-specific ones. The meta one is the one people pull most of the time.

**Review by mx-psi — CHANGES_REQUESTED** [review]
This could be quite disruptive to users if they are not aware of the current situation, since they may face a lot of breaking changes.

We should do this only for `otel/opentelemetry-collector` (and the associated one on the ghcr.io registry) on this first PR, evaluate the damage done, and do contrib on a different release.

**Comment by codeboten** [comment]
superseeded by https://github.com/open-telemetry/opentelemetry-collector-releases/pull/220

**Pull request closed without merging**